drivers/hmis/spec/requests/hmis/update_form_definition_spec.rb

```diff
@@ -0,0 +1,66 @@
+    }
+    _response, result = post_graphql(id: 30, input: input) { mutation }
+    expect(result.dig('errors')).to be_empty
+    expect(result.dig('data', 'updateFormDefinition', 'errors')).to be_empty
+  end
+
+  it 'should work when the definition does not need to be converted' do
+    input = {
+      definition: '{"item":[{"item":[{"text":"Assessment Date","type":"DATE","hidden":false,"link_id":"link_1","prefill":false,"repeats":false,"required":false,"read_only":false,"warn_if_empty":false,"assessment_date":true,"enable_behavior":"ANY","disabled_display":"HIDDEN"},{"text":"Emergency Contact Name","type":"STRING","hidden":false,"link_id":"emergency_contact_name","mapping":{"custom_field_key":"emergency_contact_name"},"prefill":false,"repeats":false,"required":false,"read_only":false,"warn_if_empty":false,"enable_behavior":"ANY","disabled_display":"HIDDEN"}],"text":"SPDAT","type":"GROUP","hidden":false,"link_id":"event_group","prefill":false,"repeats":false,"required":false,"read_only":false,"warn_if_empty":false,"enable_behavior":"ANY","disabled_display":"HIDDEN"}]}',
+    }
+    _response, result = post_graphql(id: 30, input: input) { mutation }
+    expect(result.dig('errors')).to be_empty
```

Comment:
I have found `expect(response.status).to eq(200), result.inspect` is a nice approach, it will log the result if the response is not 200:

```
     Failure/Error: expect(response.status).to eq(200), result.inspect
       {"errors"=>[{"message"=>"not found", "backtrace"=>["/app/drivers/hmis/app/graphql/mutations/update_form_definition.rb:18:in `resolve'", "/bundle/ruby/3.1.0/gems/graphql-2.3.1/lib/graphql/schema/resolver.rb:119:in `public_send'", "/bundle/ruby/3.1.0/gems/graphql-2.3.1/lib/graphql/schema/resolver.rb:119:in `call_resolve'", "/bundle/ruby/3.1.0/gems/graphql-2.3.1/lib/graphql/schema/mutation.rb:69:in `call_resolve'
...
```

It's not finding a definition with id `30`. You'll need to create a definition using the `hmis_form_definition` factory

Reply:
Thank you. 🙏 did you find this by running it locally? It succeeded for me locally and I'm curious about why that is -- even when I added 
```
  before(:all) do
    cleanup_test_environment
  end
```

Reply:
Probably because the `fd1` fixture happened to have database id `30` for you locally. I'm guessing thats how you came up with the number 30 in the first place :)

The IDs on those fixtures will not be consistent.
One way you can clear out the forms in your test db is `rails c -e test`, `Hmis::Form::Definiton.delete_all`.